File: schemas/digital-asset-management/convex/tags.ts
// tags: Workspace-scoped labels for categorizing and searching assets.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const tags = defineTable({
  workspaceId: v.id("workspaces"),
  name: v.string(),
})
  .index("by_workspace_id_and_name", ["workspaceId", "name"]);

File: schemas/digital-asset-management/spacetimedb/tags.rs
// tags: Workspace-scoped labels for categorizing assets.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

// Composite unique(workspace_id, name) — enforce in reducer logic

#[spacetimedb::table(name = tags, public)]
pub struct Tag {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub workspace_id: String, // FK → workspaces.id (cascade delete)

    pub name: String,

    pub created_at: Timestamp,
}
